cms/search/signal_handlers.py

```diff
@@ -28,7 +28,10 @@ def on_page_published(sender: "type[Page]", instance: "Page", **kwargs: Any) ->
     """Called whenever a Wagtail Page is published (UI or code).
     instance is the published Page object.
     """
-    if instance.specific_class.__name__ not in settings.SEARCH_INDEX_EXCLUDED_PAGE_TYPES:
+    if (
+        instance.specific_class.__name__ not in settings.SEARCH_INDEX_EXCLUDED_PAGE_TYPES
+        and not instance.get_view_restrictions().exists()
+    ):
         get_publisher().publish_created_or_updated(instance)
```

Comment:
I think one of the missing scenarios is, what if a page is published (without view restriction) and then the page is updated to include a view restriction?

Currently, this would be ignored as it would have a view restriction, but search results would still include private pages.

I know we decided to use `page_slug_changed` to avoid fetching the previous published version, but I believe we need to do that so we can handle the case where, if a published page that was public becomes private, we need to update the search content deleted topic to request removal.

So, on publish, if the page has a view restriction but the previous published version didn't, we should call `get_publisher().publish_deleted()` for the instance and its child that are live.